scipy/optimize/_linprog_util.py

```diff
@@ -147,80 +203,55 @@ def _clean_inputs(c, A_ub=None, b_ub=None, A_eq=None, b_eq=None, bounds=None,
             raise ValueError(
                 "Invalid input for linprog: c must not contain values "
                 "inf, nan, or None")
-    except TypeError:
-        raise TypeError(
-            "Invalid input for linprog: c must be a 1D array of numerical "
-            "coefficients")
 
+    sparse_lhs = sps.issparse(A_eq) or sps.issparse(A_ub)
     try:
-        try:
-            if sps.issparse(A_eq) or sps.issparse(A_ub):
-                A_ub = sps.coo_matrix(
-                    (0, n_x), dtype=float) if A_ub is None else sps.coo_matrix(
-                    A_ub, dtype=float).copy()
-            else:
-                A_ub = np.zeros(
-                    (0, n_x), dtype=float) if A_ub is None else np.asarray(
-                    A_ub, dtype=float).copy()
-        except BaseException:
-            raise TypeError
+        A_ub = _format_A_constraints(A_ub, n_x, sparse_lhs=sparse_lhs)
+    except ValueError:
+        raise TypeError(
+            'Invalid value for linprog: A_ub must be a 2D array_like '
```

Comment:
Let's keep "array" or "array-like" consistent. I say might as well tell the user to make it an array.
"Invalid input for linprog: b_ub must be a 1D array numerical values."

Reply:
done.